native/tests/src/test_persistence.cpp

```diff
@@ -2,6 +2,7 @@
 #include <mixpanel/mixpanel.hpp>
 #include <mixpanel/detail/persistence.hpp>
 #include <thread>
+#include <fstream>
```

Comment:
I prefer not to expose it if we don't have to, there's a few reasons for this, but I would definitely consider it a best practice to `#include` as few headers as possible in each `hpp` file.